chain/network/src/peer/peer_actor.rs

```diff
@@ -937,95 +937,105 @@ impl PeerActor {
+                            transaction,
+                            is_forwarded: true,
+                            check_only: false,
+                        })
+                        .await
+                        .ok();
+                    None
+                }
+                RoutedMessageBody::PartialEncodedChunkRequest(request) => {
+                    network_state.shards_manager_adapter.send(
+                        ShardsManagerRequestFromNetwork::ProcessPartialEncodedChunkRequest {
+                            partial_encoded_chunk_request: request,
+                            route_back: msg_hash,
+                        },
+                    );
+                    None
+                }
+                RoutedMessageBody::PartialEncodedChunkResponse(response) => {
+                    network_state.shards_manager_adapter.send(
+                        ShardsManagerRequestFromNetwork::ProcessPartialEncodedChunkResponse {
+                            partial_encoded_chunk_response: response,
+                            received_time: clock.now().into(),
+                        },
+                    );
+                    None
+                }
+                RoutedMessageBody::VersionedPartialEncodedChunk(chunk) => {
+                    network_state
+                        .shards_manager_adapter
+                        .send(ShardsManagerRequestFromNetwork::ProcessPartialEncodedChunk(chunk));
+                    None
+                }
+                RoutedMessageBody::PartialEncodedChunkForward(msg) => {
+                    network_state.shards_manager_adapter.send(
+                        ShardsManagerRequestFromNetwork::ProcessPartialEncodedChunkForward(msg),
+                    );
+                    None
+                }
+                RoutedMessageBody::ChunkStateWitness(witness) => {
+                    network_state
+                        .client
+                        .send_async(ChunkStateWitnessMessage { witness, peer_id })
+                        .await
+                        .ok();
+                    None
+                }
+                RoutedMessageBody::ChunkEndorsement(endorsement) => {
+                    network_state
+                        .client
+                        .send_async(ChunkEndorsementMessage(endorsement))
+                        .await
+                        .ok();
+                    None
+                }
+                body => {
+                    tracing::error!(target: "network", "Peer received unexpected message type: {:?}", body);
+                    None
+                }
+            })
+        };
+        // DO NOT turn this into a scoped .entered() span, because that is incompatible
+        // with async fns.
+        result
+            .instrument(tracing::trace_span!(
+                target: "network",
+                "receive_routed_message",
+                "type" = body_type,
```

Comment:
In this case `#[instrument]` application would need to use an initially empty field and then record it later as demonstrated in the [last example here](https://docs.rs/tracing/latest/tracing/attr.instrument.html#examples-1).